asciidoc/courses/llm-fundamentals/modules/1-introduction/lessons/2-hallucination/lesson.adoc

```diff
@@ -0,0 +1,129 @@
+
+These manifestations arise from a mix of overfitting, biases in the training data, and the model's attempt to generalize from the vast information it has been trained on.
+
+
+
+// What?
+== Common Hallucination Problems
+
+Let's take a closer look at some of the reasons why this may occur.
+
+=== Temperature
+
+LLMs are configured with a temperature, that corresponds to the amount of randomness that the underlying model should use when generating the text.
+The higher the temperature value, the more random the generated result will become, and the more likely the response will contain unfactual statements.
+
+A higher temperature may be appropriate when configuring an LLM to respond with more diverse and creative outputs, but it comes at the expense of consistency and precision.
+
+For example, a higher temperature may be good when constructing a work of fiction or a novel joke.
+On the other hand, a lower temperature, even `0`, is required when a response grounded in facts is important.
+
+[TIP]
+.Consider the correct temperature
+====
+In June 2023, two US lawyers were link:https://www.reuters.com/legal/new-york-lawyers-sanctioned-using-fake-chatgpt-cases-legal-brief-2023-06-22/[sanctioned for submitting an LLM-generated legal brief] that contained six fictitious case citations.
+====
+
+A quick fix _may_ be to reduce the temperature.  But more likely, the LLM is hallucinating because it hasn't got the information required.
+
+
+=== Model Training and Complexity
+
+Large Language Models (LLMs) are often considered "black boxes" due to the difficulty in deciphering their decision-making processes. The complexity of these models combined with potential training on erroneous or misleading data means that their outputs can sometimes be unpredictable or inaccurate.
+
+For example, if asked about a controversial historical event, an LLM might produce an answer that's biased or misrepresentative, largely due to the data it was trained on. Furthermore, it would be near impossible to transparently trace back how the model arrived at that conclusion.
+
+
+
+=== Missing Information
+
+The training process for LLMs is intricate and time-intensive, often requiring vast datasets compiled over extended periods. As such, these models might lack the most recent information or might miss out on specific niche topics not well-represented in their training data.
+
+For instance, if an LLM's last update was in September 2022, it would be unaware of world events or advancements in various fields that occurred post that date, leading to potential gaps in its knowledge or responses that seem out of touch with current realities.
+
+If the user is asking a question on information that is hard to find, or outside of the public domain, it will be next to impossible for an LLM to provide an accurate response.
```

Comment:
and then it starts to create fictious information because it has been fine-tuned (e.g. with reinforcement learning with human feedback - RLHF) and set up to be helpful at all costs, instead of just conveying that it cannot answer the question.